Source/Fuse.Scripting.JavaScript/JSTest.uno

```diff
@@ -0,0 +1,75 @@
+using Uno;
+using Uno.UX;
+using Uno.Testing;
+using Uno.Compiler;
+using Uno.Collections;
+using Uno.Text;
+using Uno.Threading;
+
+using Fuse.Reactive;
+using Fuse.Scripting;
+
+namespace Fuse.Scripting.JavaScript.Test
+{
+	// A Copy of Testing's Assert that dispatched to the JS thread, caches the results and can
+	// wait safely until the results are ready
+	public class JSTest
+	{
+		readonly ManualResetEvent _testComplete = new ManualResetEvent(false);
+		readonly Action<Fuse.Scripting.Context> _test;
+		readonly List<Exception> _failures = new List<Exception>();
+
+		public JSTest(Action<Fuse.Scripting.Context> test)
+		{
+			Fuse.Reactive.JavaScript.EnsureVMStarted();
+			_test = test;
+			Fuse.Reactive.JavaScript.Worker.Invoke(TestBodyWrapper);
+		}
+
+		void AddTestToJSContext(Fuse.Scripting.Context context)
+		{
+			var f = context.Evaluate("", "(function(obj, assert) { obj['test'] = { assert: function(exp, msg) { try { assert(Boolean(exp ? 1 : 0), msg); } catch(e) { assert(0, 'Error: ' + e); } } }; } )") as Fuse.Scripting.Function;
```

Comment:
`as Fuse.Scripting.Function` is not the appropriate cast here, if it fails we're just going to crash with an NRE on the next line. Please do
```uno
var f = (Fuse.Scripting.Function)context.Evaluate("", "(function(obj, assert) { obj['test'] = { assert: function(exp, msg) { try { assert(Boolean(exp ? 1 : 0), msg); } catch(e) { assert(0, 'Error: ' + e); } } }; } )");
```
instead.

Reply:
OK, seems you're just moving code here, so I guess that'd be an inappropriate time to sneak in a change like that. I'll just clean this up afterwards instead.

Reply:
👍